examples/e2e/steps/inference/inference_predict.py

```diff
@@ -0,0 +1,62 @@
+#  Copyright (c) ZenML GmbH 2023. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+
+from typing import Annotated
+
+import pandas as pd
+from config import MetaConfig
+
+from zenml import step
+from zenml.integrations.mlflow.model_deployers.mlflow_model_deployer import (
+    MLFlowDeploymentService,
+)
+from zenml.integrations.mlflow.steps.mlflow_deployer import (
+    mlflow_model_registry_deployer_step,
+)
+
+
+@step
+def inference_predict(
+    dataset_inf: pd.DataFrame, model_version: str
+) -> Annotated[pd.Series, "predictions"]:
+    """Predictions step.
+
+    This is an example of a predictions step that takes the data in and returns
+    predicted values.
+
+    This step is parameterized, which allows you to configure the step
+    independently of the step code, before running it in a pipeline.
+    In this example, the step can be configured to use different input data
+    and model version in registry. See the documentation for more information:
+
+        https://docs.zenml.io/user-guide/advanced-guide/configure-steps-pipelines
+
+    Args:
+        dataset_inf: The inference dataset.
+        model_version: Model Version in Model Registry.
+
+    Returns:
+        The processed dataframe: dataset_inf.
+    """
+    ### ADD YOUR OWN CODE HERE - THIS IS JUST AN EXAMPLE ###
+    deployment: MLFlowDeploymentService = mlflow_model_registry_deployer_step(
```

Comment:
Wait, does this work?

Reply:
Anyways, maybe nicer to split this up by moving `mlflow_model_registry_deployer_step` invocation into the pipeline itself and passing the resulting service into this step as an input

Reply:
Yes, it also works. Moving this out would be fine here, but might be a bit complex for the promotion stage. I'll think about it a bit more.

Reply:
Carved out